Compile error on latest Alpine

PowerDNS gives me compile errors for 4.1.2 and 4.1.4 on latest alpine docker base image.

```
checking whether compiling and linking against OpenSSL's libcrypto works... yes
checking /usr/include/openssl/ecdsa.h usability... yes
checking /usr/include/openssl/ecdsa.h presence... yes
checking for /usr/include/openssl/ecdsa.h... yes
```

```
CXX      opensslsigners.o
opensslsigners.cc: In function 'void RSA_get0_key(const RSA*, const BIGNUM**, const BIGNUM**, const BIGNUM**)':
opensslsigners.cc:80:20: error: 'void RSA_get0_key(const RSA*, const BIGNUM**, const BIGNUM**, const BIGNUM**)' was declared 'extern' and later 'static' [-fpermissive]                                                                                                                                                           
 static inline void RSA_get0_key(const RSA* rsakey, const BIGNUM** n, const BIGNUM** e, const BIGNUM** d) {                                                      
                    ^~~~~~~~~~~~                                                                           

....

opensslsigners.cc: In function 'int ECDSA_SIG_set0(ECDSA_SIG*, BIGNUM*, BIGNUM*)':
opensslsigners.cc:137:19: error: 'int ECDSA_SIG_set0(ECDSA_SIG*, BIGNUM*, BIGNUM*)' was declared 'extern' and later 'static' [-fpermissive]
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                   ^~~~~~~~~~~~~~
In file included from opensslsigners.cc:27:0:
/usr/include/openssl/ecdsa.h:148:5: note: previous declaration of 'int ECDSA_SIG_set0(ECDSA_SIG*, BIGNUM*, BIGNUM*)'
 int ECDSA_SIG_set0(ECDSA_SIG *sig, BIGNUM *r, BIGNUM *s);
     ^~~~~~~~~~~~~~
make[3]: *** [Makefile:2581: opensslsigners.o] Error 1
make[3]: Leaving directory '/tmp/pdns-4.1.2/pdns'
make[2]: *** [Makefile:2145: all] Error 2
make[2]: Leaving directory '/tmp/pdns-4.1.2/pdns'
make[1]: *** [Makefile:540: all-recursive] Error 1
make[1]: Leaving directory '/tmp/pdns-4.1.2'
make: *** [Makefile:471: all] Error 2
```

To reproduce you could checkout master from https://github.com/psi-4ward/docker-powerdns and run a `docker build .`

Probably a bug with libressl (v2.7)?

Thanks for your help!

--- a/pdns/opensslsigners.cc
+++ b/pdns/opensslsigners.cc
@@ -77,6 +77,8 @@ void openssl_thread_cleanup()
   OPENSSL_free(openssllocks);
 }
 
+#if !defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x2070000fL
+/* those symbols are defined in LibreSSL 2.7.0+ */
 /* compat helpers. These DO NOT do any of the checking that the libssl 1.1 functions do. */
 static inline void RSA_get0_key(const RSA* rsakey, const BIGNUM** n, const BIGNUM** e, const BIGNUM** d) {
   *n = rsakey->n;
@@ -143,6 +145,8 @@ static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
   return 1;
 }
 #endif /* HAVE_LIBCRYPTO_ECDSA */
+
+#endif /* !defined(LIBRESSL_VERSION_NUMBER) || LIBRESSL_VERSION_NUMBER < 0x2070000fL */
 
 #else
 void openssl_thread_setup() {}

